tools/autograd/gen_python_functions.py

```diff
@@ -759,8 +759,12 @@ def sort_overloads(
 ) -> Sequence[PythonSignatureGroup]:
 
     def is_arg_smaller(t1: Type, t2: Type) -> bool:
+        # int/int? has been given higher priority than int[] here, because argument parser
+        # perceives integer inputs as int[] when the number of arguments is equal to 1 and 
+        # signature with int[] is located earlier than other signatures list.
```

Comment:
I think you can do a little better on this comment. What you have described here is the "what", as in, if you didn't setup the ordering here, this is WHAT would have happened. But comments should describe the "why". WHY is it bad if you don't reorder the parameters here?

There's plenty of discussion in the earlier PR which you can use as the basis for explaining the why.

Reply:
Thanks for pointing out, and the comment changed accordingly :)